### PR TITLE
Add a simple paraview module

### PR DESCRIPTION
**Pull request by vollia**
Hello @rupertnash ,

This PR add a simple ParaView 5.9.0 module with a pvserver on it.
I will now create another module, named paraview-ttk, that will use the ParaView Superbuild branch I created with TTK.

Best,
Charles

### CODE REVIEW

File: packages/paraview/env.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+if [ -z "$INSTALLER_PARAVIEW_ENV_SH" ]; then
+    INSTALLER_PARAVIEW_ENV_SH=1
+    installer_paraview_dir=$(readlink -f $(dirname $BASH_SOURCE))
+    . $installer_paraview_dir/../../env.sh
+
+    # modules
+    module load cmake
+
+    # Closed source, keep VESTEC only
+    visibility=private
```
Comment (rupertnash) [inline packages/paraview/env.sh]: Is there any reason this is private?

Comment (vollia) [reply packages/paraview/env.sh]: it can be public indeed
```suggestion
    visibility=public
```
just like that ?

**Comment by rupertnash** [comment]
For both 5.8 and 5.9, when I try to use use via CMake, I'm told the installed version is `5.6.0-5602-g15a2427cb0`. E.g.:
```
cmake /home/d170/d170/nashvest/app-install/packages/fireapp/build/fireapp -DCONAN_CMD=/work/d170/shared/sw/conan/1.31.3/bin/conan -DUSE_CATALYST=ON -DCMAKE_BUILD_TYPE=Release -DFIREAPP_CONAN_GDAL=OFF -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=/work/d170/d170/shared/sw/fireapp/CatalystPV -DFIREAPP_CONAN=ON
-- The C compiler identification is GNU 10.1.0
<snip />
CMake Error at Adaptor/CMakeLists.txt:1 (find_package):
  Could not find a configuration file for package "ParaView" that is
  compatible with requested version "5.8".

  The following configuration files were considered but not accepted:

    /work/d170/d170/shared/sw/paraview/v5.8.0/lib64/cmake/paraview-5.6/paraview-config.cmake, version: 5.6.0-4743-ge2ef711187



-- Configuring incomplete, errors occurred!
See also "/home/d170/d170/nashvest/app-install/packages/fireapp/build/build-CatalystPV/CMakeFiles/CMakeOutput.log".
```

**Comment by vollia** [comment]
Hello @rupertnash 
I have updated this package and it now build and install a PV 5.9. I was able to build the WFA app against this ParaView.

I would like to ensure the installed ParaView can be used by anyone. Would you have some time to test this ?

Best,
Charles

**Comment by vollia** [comment]
Oh, I need to rebase my change, I am on it to clean this

**Comment by rupertnash** [comment]
Great - let me know when to look at it!

**Comment by vollia** [comment]
I have rebased and squashed to keep one commit for ParaView and another for the WFA updates (can be in another MR if you want).
I have re-run this on my side successfully. It is ready to test :)

File: packages/fireapp/env.sh [hunk 2]
```diff
@@ -19,4 +19,6 @@ if [ -z "$INSTALLER_FIREAPP_ENV_SH" ]; then
     cmake_vars[FIREAPP_CONAN]=ON
     cmake_vars[FIREAPP_CONAN_GDAL]=OFF
     cmake_vars[CONAN_CMD]=$CONAN_DIR/bin/conan
+    cmake_vars[USE_CATALYST]=ON
+    cmake_vars[ParaView_DIR]=/lus/cls01095/work/d170/shared/sw/paraview/v5.9.0/lib/cmake/paraview-5.9/
```
Comment (rupertnash) [inline packages/fireapp/env.sh]: Please don't hard code machine specific paths. Should probably load the paraview module and then use e.g. `$PARAVIEW_DIR/lib/cmake/paraview-$PARAVIEW_VERSION` (making sure the paraview module defines those variables)

File: packages/paraview/env.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    declare -A cmake_vars
+    cmake_vars[BUILD_TESTING]=OFF
+    cmake_vars[PARAVIEW_BUILD_SHARED_LIBS]=ON
+    cmake_vars[PARAVIEW_BUILD_EDITION]=CATALYST_RENDERING
+    cmake_vars[paraview_SOURCE_SELECTION]=$version_num
+    cmake_vars[ENABLE_boost]=ON
+    cmake_vars[ENABLE_protobuf]=ON
+    cmake_vars[ENABLE_python]=ON
+    cmake_vars[ENABLE_python3]=ON
+    cmake_vars[USE_SYSTEM_python3]=OFF
```
Comment (rupertnash) [inline packages/paraview/env.sh]: Any reason we can't use an existing python?

File: packages/fireapp/download.sh [hunk 4]
```diff
@@ -9,8 +9,8 @@ thisdir=$(readlink -f $(dirname $BASH_SOURCE))
 
 if [ -d fireapp ]; then
     pushd fireapp
-    git pull --all --ff-only origin $version
+    git pull --ff-only origin Catalyst
```
Comment (rupertnash) [inline packages/fireapp/download.sh]: I'd prefer to leave this as `version` rather than hardcoding.

File: packages/fireapp/download.sh [hunk 5]
```diff
@@ -9,8 +9,8 @@ thisdir=$(readlink -f $(dirname $BASH_SOURCE))
...
     popd
 else
-    git clone -b $version https://git.firesponse.es/external/vestec.git fireapp
+    git clone -b Catalyst https://git.firesponse.es/external/vestec.git fireapp
```
Comment (rupertnash) [inline packages/fireapp/download.sh]: ditto

File: packages/fireapp/env.sh [hunk 6]
```diff
@@ -6,7 +6,7 @@ if [ -z "$INSTALLER_FIREAPP_ENV_SH" ]; then
     # Closed source, keep VESTEC only
     visibility=private
     name=fireapp
-    version=Catalyst
+    version=Kitware_Catalyst
```
Comment (rupertnash) [inline packages/fireapp/env.sh]: revert this please - once this all works your build will replace the previous one

File: packages/paraview/install.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+if [[ "${BASH_SOURCE[0]}" != "${0}" ]]; then
+    echo "Don't source me"
+    return 1
+fi
+set -e
+thisdir=$(readlink -f $(dirname $BASH_SOURCE))
+. $thisdir/env.sh
+
+pushd $build_dir_name
+# do not use make install so we also access the sdk
```
Comment (rupertnash) [inline packages/paraview/install.sh]: What does this mean?

File: packages/paraview/env.sh (hunk 3 above)
Comment (vollia) [reply packages/paraview/env.sh]: The existing Python I found (3.8.5) make the compilation fails due to a symbol error. (It misses the `-fPic` flag if I remember correctly)
I will remove the module Python for this build

File: packages/paraview/install.sh (hunk 7 above)
Comment (vollia) [reply packages/paraview/install.sh]: if we use `make install` the sdk files (especially lib/cmake/*) are not installed. This means we cannot do a `find_package` to retrieve the ParaView configuration.

**Comment by vollia** [comment]
I have addressed your remarks and changed the way `$version` is done to be consistent with other modules

File: packages/paraview/install.sh (hunk 7 above)
Comment (rupertnash) [reply packages/paraview/install.sh]: Roger that

**Comment by rupertnash** [comment]
Hi @CharlesGueunet - I think I got this working on Cirrus at least. Could you please try on ARCHER2 and if that's good we can merge!

**Comment by vollia** [comment]
I have build and installed both ParaView and Fireapp successfully on ARCHER2

**Pull request merged**